exist-core/src/main/java/org/exist/xquery/GroupByClause.java

```diff
@@ -185,11 +189,31 @@ public Sequence postEval(final Sequence seq) throws XPathException {
     public void analyze(AnalyzeContextInfo contextInfo) throws XPathException {
         contextInfo.setParent(this);
         unordered = (contextInfo.getFlags() & UNORDERED) > 0;
+
+        final Set<QName> preGroupingTupleVariables = new HashSet<>();
+        FLWORClause prevClause = getPreviousClause();
+        while (prevClause != null) {
+            rootClause = prevClause;
+            preGroupingTupleVariables.addAll(prevClause.getTupleStreamVariables());
+            prevClause = prevClause.getPreviousClause();
+        }
+
         final LocalVariable mark = context.markLocalVariables(false);
         try {
             if (groupSpecs != null) {
                 for (final GroupSpec spec : groupSpecs) {
-                    final LocalVariable groupKeyVar = new LocalVariable(spec.getKeyVarName());
+                    final QName keyVarName = spec.getKeyVarName();
+
+                    if (spec.getGroupExpression() instanceof VariableReference) {
+                        final VariableReference groupExpressionVariableRef = ((VariableReference) spec.getGroupExpression());
+                        if (groupExpressionVariableRef.getName().equals(keyVarName)) {
+                            if (!preGroupingTupleVariables.contains(keyVarName)) {
```

Comment:
Any reason not joining this expressions?

Reply:
Done.